File: quicksort-34b.py
def quicksort(arr):
    if len(arr) <= 1:
        return arr
    pivot = arr[0]
    less = []
    greater = []
    for x in arr[1:]:
        if x < pivot:
            less.append(x)
        else:
            greater.append(x)
    return quicksort(less) + [pivot] + quicksort(greater)

File: quicksort-34b.rb
  def quicksort(ary)
    return [] if ary.size <= 1
    pivot = ary.delete_at(rand(ary.size))
    less = quicksort(ary.select { |x| x <= pivot })
    greater = quicksort(ary.select { |x| x > pivot })
    less + [pivot] + greater
  end

